src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/LabelSelectorExpressions.java

```diff
@@ -0,0 +1,83 @@
+package com.cloudbees.jenkins.plugins.kubernetes_credentials_provider;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.api.model.LabelSelectorBuilder;
+
+/**
+ * Parser for <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors">Kubernetes Label Selectors</a>.
+ */
+class LabelSelectorExpressions {
+
+    /**
+     * Parse Kubernetes label selector expression.
+     * Example:
+     * <pre>
+     * app in (jenkins-dev, jenkins-all)
+     * </pre>
+     * @param selector label selector expression or null
+     * @return parsed label selector
+     */
+    @NonNull
+    static LabelSelector parse(String selector) {
+        LabelSelectorBuilder lsb = new LabelSelectorBuilder();
+        if (selector != null && !selector.trim().isEmpty()) {
+            Matcher matcher = Pattern.compile("[^,]+?\\([^()]+\\)|[^,]+").matcher(selector);
+            while (matcher.find()) {
+                String expression = matcher.group();
+                expression = expression.replaceFirst("!=|=", " $0 ");
+                String[] tokens = expression.trim().split("\\s+", 3);
+                switch (tokens.length) {
```

Comment:
introduces a spotbugs warning.

```
[2019-07-19T21:28:02.599Z] [INFO] Switch statement found in com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.LabelSelectorExpressions.parse(String) where default case is missing [com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.LabelSelectorExpressions] At LabelSelectorExpressions.java:[lines 33-71] SF_SWITCH_NO_DEFAULT
```

Reply:
merged `case 2` into the required `default` case which now throws a parser exception